### DISCUSSION TITLE
How do I increase the interval between syncs?

### DISCUSSION THREAD

**Discussion by sitadrost** [abraunegg/onedrive#1518] [Q&A] [answered]
I'm using the onedrive client to back up the local data on my Ubuntu 20.04 machine on my OneDrive. As I installed the client according to the instructions in INSTALL.md, it is automatically running in monitor mode. As far as I understood from the docs, in monitor mode the client syncs every `monitor_interval` seconds.

At the moment I'm running simulations, the results of which are written to a couple of text files. Because these simulations typically take around one to several hours to run, I tried setting the `monitor_interval` value to 6 hours (see below), to prevent new versions of my results text files being uploaded to OneDrive unnecessarily often.

However, I'm afraid I did something wrong somewhere, because every time I run a simulation, the Recycle Bin on my OneDrive still fills up with heaps of deleted result text files. They are time stamped 1 minute apart, so on second thought the `monitor_interval` doesn't seem to have an awful lot to do with this.

Does anyone know what causes this behaviour, and how to prevent this?

Many thanks!
Sita

```
onedrive version                       = v2.4.10-0~202102221319~ubuntu20.04.1
Config path                            = /home/sita/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/sita/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 21600
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```

**Comment by abraunegg** [comment, marked as answer]
@sitadrost 
There are a number of issues here:

1. You are using an unsupported client. Whilst you are using the Ubuntu PPA, it is out of date and contains defects. You need to uninstall the PPA package and build from source.
2. The Ubuntu PPA package is impacted by this issue: https://github.com/abraunegg/onedrive/issues/1274
3. Because of point 2, you have the client running in --monitor mode (even if you did not activate any systemd servcice), meaning, it is acting on local writes to your 'sync_dir' immediately. The  'monitor_interval' function is how frequently checks are made of new changes on OneDrive online - not local changes. If does not govern local -> OneDrive online file uploads (which is what you are seeing).

To resolve your issue:
1. Uninstall the unsupported Ubuntu PPA package
2. Build and install from source
3. Do not activate any systemd service
4. Sync manually post running your simulations using `--synchronize`

If you *must* run the systemd service for other reasons, when running your simulations, first stop the onedrive service before starting your simulations, then restart upon completion.

**Comment by sitadrost** [comment]
@abraunegg
Thanks for your quick response!
Ah, that explains, I thought the monitor interval would work both ways, but seeing as that isn’t the case, it’s indeed a good idea not to run the client in monitor mode during simulations.

So the client automatically running in monitor mode is a bug. Alright, I’ll uninstall and re-install from source this time. How does that work with updates, do I need to check for new versions every now and then?

**Reply by abraunegg** [reply]
@sitadrost 
>  How does that work with updates, do I need to check for new versions every now and then?

Yes you will need to rebuild every month or so. Best way to keep track is to watch the repo so you get notified of new releases.

Alternative actions are:
* Raise the package version with the package maintainer and get them to update / kept updated
* Build your own package and update that in an automated fashion when new releases are done
* Switch your Linux distribution to something other than Ubuntu, where this package  is kept more up to date
